Configured directory not existing causes panic with non-informative error message

### What happened?

Trying to use local.Command to upload assets to s3 but have been encountering this issue for a while. This *has* worked in the past - no pinpoint on when this started failing yet (any chance it could be a node bump from v14 -> v16?).

```
panic: runtime error: invalid memory address or nil pointer dereference
    [signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x102de34a0]
    goroutine 20 [running]:
    github.com/pulumi/pulumi-command/provider/pkg/provider.(*command).run(0x14000156230, {0x1030e6d60, 0x1400012ea40}, {0x1400012c5a0, 0x57}, 0x1400000d200, {0x14000162150, 0x67})
```

### Steps to reproduce

Using `local.Command` in the following way to upload static site assets to s3. Note the `'echo "skipped sync"'` commented variant of the command too which I've tested and found also fails - so validating its not an issue with the specific command being run.

```
const syncCredentials = pulumi.output(assumePulumiRole('sync-s3'))
const awsS3SyncCommand = pulumi.interpolate`aws s3 sync ./ s3://${bucket.bucket} --cache-control "public, max-age=3600;"`
// const awsS3SyncCommand = 'echo "skipped sync"'

// ensure all static assets are uploaded
const sync = new cmd.local.Command(
  resourceName([sttic.name, 'sync'], parent),
  {
    create: awsS3SyncCommand,
    update: awsS3SyncCommand,
    dir: getProjectPath(sttic.ctx.project!, sttic.args.path),
    environment: {
      AWS_REGION: config.region,
      AWS_SESSION_TOKEN: syncCredentials.sessionToken.apply((v) => v!),
      AWS_ACCESS_KEY_ID: syncCredentials.accessKeyId.apply((v) => v!),
      AWS_SECRET_ACCESS_KEY: syncCredentials.secretAccessKey.apply(
        (v) => v!
      ),
    },
  },
  { parent: this }
)

sync.stderr.apply((output) => {
  pulumi.log.info(`stderr of command: ${awsS3SyncCommand}`)
  pulumi.log.warn(output)
})
sync.stdout.apply((output) => {
  pulumi.log.info(`stdout of command: ${awsS3SyncCommand}`)
  pulumi.log.info(output)
})
```

### Expected Behavior

`Command` runs successfully and s3 assets are uploaded.

### Actual Behavior

```
  error: update failed
 
    panic: runtime error: invalid memory address or nil pointer dereference
    [signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x102de34a0]
    goroutine 20 [running]:
    github.com/pulumi/pulumi-command/provider/pkg/provider.(*command).run(0x14000156230, {0x1030e6d60, 0x1400012ea40}, {0x1400012c5a0, 0x57}, 0x1400000d200, {0x14000162150, 0x67})
        /home/runner/work/pulumi-command/pulumi-command/provider/pkg/provider/command.go:136 +0x6d0
    github.com/pulumi/pulumi-command/provider/pkg/provider.(*command).RunCreate(0x14000156230, {0x1030e6d60, 0x1400012ea40}, 0x1400000d200, {0x14000162150, 0x67})
        /home/runner/work/pulumi-command/pulumi-command/provider/pkg/provider/command.go:53 +0x60
    github.com/pulumi/pulumi-command/provider/pkg/provider.(*commandProvider).Create(0x14000067680, {0x1030e6e08, 0x14000128e10}, 0x14000134780)
        /home/runner/work/pulumi-command/pulumi-command/provider/pkg/provider/provider.go:213 +0x8c8
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Create_Handler.func1({0x1030e6e08, 0x14000128e10}, {0x10305b780, 0x14000134780})
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.35.1/proto/go/provider.pb.go:3723 +0x7c
    github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1({0x1030e6e08, 0x14000128ab0}, {0x10305b780, 0x14000134780}, 0x14000124360, 0x140001324c8)
        /home/runner/go/pkg/mod/github.com/grpc-ecosystem/grpc-opentracing@v0.0.0-20180507213350-8e809c8a8645/go/otgrpc/server.go:57 +0x394
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Create_Handler({0x10309cca0, 0x14000067680}, {0x1030e6e08, 0x14000128ab0}, 0x1400011c840, 0x14000031b40)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.35.1/proto/go/provider.pb.go:3725 +0x150
    google.golang.org/grpc.(*Server).processUnaryRPC(0x140002f68c0, {0x1030f4590, 0x1400031b1e0}, 0x14000152240, 0x140002ef4a0, 0x10366a360, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.45.0/server.go:1282 +0xc5c
    google.golang.org/grpc.(*Server).handleStream(0x140002f68c0, {0x1030f4590, 0x1400031b1e0}, 0x14000152240, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.45.0/server.go:1619 +0xa34
    google.golang.org/grpc.(*Server).serveStreams.func1.2(0x1400038e010, 0x140002f68c0, {0x1030f4590, 0x1400031b1e0}, 0x14000152240)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.45.0/server.go:921 +0x94
    created by google.golang.org/grpc.(*Server).serveStreams.func1
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.45.0/server.go:919 +0x1f0
 
  command:local:Command (pre-dev-docs-sync):
    error: error reading from server: EOF
```

### Versions used

```
CLI          
Version      3.34.1
Go Version   go1.17.11
Go Compiler  gc

Plugins
NAME    VERSION
nodejs  unknown

Host     
OS       darwin
Version  12.0.1
Arch     arm64

This project is written in nodejs: executable='/nix/store/z44kqj0qn5b8pw40fm7vd37dlcrxd58y-nodejs-16.15.0/bin/node' version='v16.15.0'
```

command v0.3.2

### Additional context

Had hoped #68 would resolve but no luck on `0.3.2`.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/provider/pkg/provider/command.go
+++ b/provider/pkg/provider/command.go
@@ -79,7 +79,6 @@ func (c *command) RunCreate(ctx context.Context, host *provider.HostClient, urn 
 	return id, err
 }
 
-//
 func (c *command) RunUpdate(ctx context.Context, host *provider.HostClient, urn resource.URN) (string, error) {
 	if c.Update != nil {
 		stdout, stderr, id, err := c.run(ctx, *c.Update, host, urn)
@@ -131,6 +130,14 @@ func (c *commandContext) run(ctx context.Context, command string, host *provider
 	cmd.Stdout = stdoutw
 	cmd.Stderr = stderrw
 	if c.Dir != nil {
+		// Check if exists and is a directory.
+		stat, err := os.Stat(*c.Dir)
+		if err != nil {
+			return "", "", "", fmt.Errorf("error looking up directory: %q", err)
+		}
+		if !stat.IsDir() {
+			return "", "", "", fmt.Errorf("%s is not a directory", *c.Dir)
+		}
 		cmd.Dir = *c.Dir
 	} else {
 		cmd.Dir, err = os.Getwd()

